packages/babel-traverse/src/index.ts

```diff
@@ -100,6 +101,17 @@ traverse.node = function (
   // traverse.node always returns undefined
 };
 
+traverse.direct = function (
+  path: NodePath,
```

Comment:
I really dislike the inconsistency that all the `traverse.*` methods take a Node, while this takes a NodePath. Can we move it to a `NodePath.prototype.traverseSelf(opts, state)` method?

Reply:
Maybe it can be an option for traverse.
`path.traverse(visitor, state, opts)`

Reply:
> I really dislike the inconsistency that all the traverse.* methods take a Node, while this takes a NodePath.

Makes sense, I've removed `traverse.direct` in 29f8d17.

> Can we move it to a NodePath.prototype.traverseSelf(opts, state) method?
> Maybe it can be an option for traverse.
path.traverse(visitor, state, opts)

Yeah, absolutely. Moving this to NodePath keeps the traverse "node centric", and extending `path.traverse` with an optional third argument (`opts`) sounds like a good idea. I've implemented that in 3e6fe18. What do you think?

Reply:
This looks fine to me, but only for the `visitSelf` feature. Regarding `Enables visiting a NodePath directly`, let's hear @nicolo-ribaudo @JLHwung 

Also I think the CI bug is related to this PR, can you take a look?

Reply:
Thanks, I'll take a look at the CI bug. 

If this new direction makes sense (`path.traverse` with optional `opts`), I can update the PR title and description accordingly.